### PR TITLE
Fix docstrings, typing and string construction - agent

### LINKED ISSUE #91: Fix docstrings, typing and string construction - agent

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #91

### CODE REVIEW

**Review by WerLaj — APPROVED** [review]
LGTM

File: moviebot/agent/agent.py [hunk 1]
```diff
@@ -125,29 +99,25 @@ def initialize(self, user_id=None):  # noqa: C901
...
+            if path:
+                self.bot_recorder = RecorderBot(path)
+            else:
+                raise ValueError("Path to save conversation is not provided.")
+
+    def start_dialogue(
+        self, user_fname: str = None, restart: bool = False
+    ) -> Union[
+        Tuple[str, DialogueOptions],
+        Tuple[str, Dict[str, Any], DialogueOptions],
```
Comment (NoB0) [inline moviebot/agent/agent.py:119]: Should we change the order in this tuple like `str, DialogueOptions, Dict[str, Any]?
This way it is easier to identify what is what based on the returns statement.

File: moviebot/agent/agent.py [hunk 2]
```diff
@@ -175,26 +146,34 @@ def start_dialogue(self, user_fname=None, restart: bool = False):
...
 
-    def continue_dialogue(self, user_utterance, user_options, user_fname=None):
+    def continue_dialogue(
+        self,
+        user_utterance: UserUtterance,
+        user_options: DialogueOptions,
+        user_fname: str = None,
+    ) -> Union[
+        Tuple[str, DialogueOptions],
+        Tuple[str, Dict[str, Any], DialogueOptions],
```
Comment (NoB0) [inline moviebot/agent/agent.py:163]: Same comment as above.

File: moviebot/dialogue_manager/dialogue_state.py [hunk 3]
```diff
@@ -61,7 +62,7 @@ def _agent_offer_state(self):
         }
         return str([x for x, y in offer_state.items() if y])
 
-    def _dict(self) -> str:
+    def to_dict(self) -> Dict[str, Any]:
         """Prints the state to debug the position of the agent.
```
Comment (NoB0) [inline moviebot/dialogue_manager/dialogue_state.py:66]: Nit: this method does not print. Maybe the Docstring can be updated to something like: `Returns the dialogue state as a dictionary.`

**Review by NoB0 — APPROVED** [review]
LGTM. See inline questions.

File: moviebot/agent/agent.py (hunk 1 above)
Comment (IKostric) [reply moviebot/agent/agent.py]: Done

File: moviebot/agent/agent.py (hunk 2 above)
Comment (IKostric) [reply moviebot/agent/agent.py]: Done

File: moviebot/dialogue_manager/dialogue_state.py (hunk 3 above)
Comment (IKostric) [reply moviebot/dialogue_manager/dialogue_state.py]: Done

**Pull request merged**